src/pages/Subscribe/Subscribe.js

```diff
@@ -1,7 +1,362 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useNavigate, useLocation } from 'react-router-dom';
 
 const Subscribe = () => {
-  return <h1>Subscribe</h1>;
+  const { pathname } = useLocation();
+
+  const [index, setIndex] = useState(1);
+  const [name, setName] = useState('');
+  const [address, setAddress] = useState('');
+  const PRICE = 39000;
+
+  const result = PRICE * index;
+
+  const navigete = useNavigate();
+
+  const goToBack = () => {
+    navigete(-1);
+  };
+
+  const onChangeName = e => {
+    setName(e.target.value);
+  };
+  const onChangeAddress = e => {
+    setAddress(e.target.value);
+  };
+
+  const onIndex = e => {
+    setIndex(e.target.value);
+  };
```

Comment:
```suggestion javascript
const getUserInfo = (e)=>{
const  {value,name} = e.target
setUserInfo({...userInfo, [name] : value})
}
```

* 같은 동작을 하는 함수를 하나로 묶어서 활용 해 보세요.
* `[name]:value` 이부분은 객체의 계산된 속성명 문법입니다.